lib/core/string.toit

```diff
@@ -244,6 +244,65 @@ abstract class string implements Comparable:
       rune := this[i]
       if rune: block.call rune
 
+  /**
+  For every unicode character in the string, call the block.
+  The argument to the block is an integer in the Unicode range of 0-0x10ffff,
+    inclusive.
+  The return value is assembled from the return values of the block.
+  If a string is returned from the block it is inserted at that point in the
+    return value.
+  If a byte array is returned from the block it is converted to a string and
+    treated like a string.  The byte array must contain whole, valid UTF-8
+    sequences.
+  If an integer is returned from the block it is treated as a Unicode code
+    point, and the corresponding code point is inserted.
+  If the block returns null, this is treated like the zero length string.
+  If the block returns a collection, then every element in the collection
```

Comment:
This isn't implemented.

Also, I think we shouldn't use `Collection` in general. (I would even vote in favor of removing it entirely).
Here, the order matters, so `List` is the only thing that would make sense.

Reply:
All our collections are ordered.